internal/cmd/gencli/templates.go

```diff
@@ -385,10 +385,11 @@ func (c *{{ camelCase .SubActionPrefix }}Command) Run(args []string) int {
 		c.FlagSecrets,
 		c.FlagScrts,
 		func() {
-			opts = append(opts, {{ .Pkg }}.DefaultSecrets())
+			{{ if (and (eq .ResourceType "host-catalog") (eq .SubActionPrefix "plugin")) }} opts = append(opts, {{ .Pkg }}.DefaultSecrets()){{ end }}
```

Comment:
I haven't gotten to the API side of things here yet (going in order on the PR) but this change strikes me as a bit eyebrow-raising -- we have avoided putting type-specific items into the templates so far...

Reply:
I agree, I don't like having this conditional check too. 

When generating the code within the api package, the DefaultSecrets() function is not generated for kv credentials because secrets is a required field. The reason behind this conditional check in the template is because dynamic host catalog secrets & kv credentials share some code when it comes to the api and cli. 
For the dynamic host catalog, secrets is an optional field, which is why it has the DefaultSecrets function. 

I could change the kv credential api input so that a default value function is created, but then it would be an anti-pattern for how our other credentials required fields behave, such as the password field in a static username & password credential.

Reply:
I replaced the string comparison conditions with the variable IsPluginType.